packages/realm/src/Configuration.ts

```diff
@@ -195,16 +197,32 @@ type BaseConfiguration = {
 //   disableFormatUpgrade?: boolean;
 // };
 
-const OBJECT_SCHEMA_KEYS = new Set<keyof ObjectSchema>(["name", "primaryKey", "embedded", "asymmetric", "properties"]);
+// Need to use `CanonicalObjectSchema` rather than `ObjectSchema` due to some
+// integration tests using `openRealmHook()`. That function sets `this.realm`
+// to the opened realm whose schema is a `CanonicalObjectSchema[]`. Consequently,
+// the key `"ctor"` (which doesn't exist on `ObjectSchema`) also needs to be allowed.
+const OBJECT_SCHEMA_KEYS = new Set<keyof CanonicalObjectSchema>([
+  "name",
+  "primaryKey",
+  "embedded",
+  "asymmetric",
+  "properties",
+  // Not part of `ObjectSchema`
+  "ctor",
+]);
 
-const PROPERTY_SCHEMA_KEYS = new Set<keyof ObjectSchemaProperty>([
+// Need to use `CanonicalObjectSchemaProperty` rather than `ObjectSchemaProperty`
+// due to the same reasons as above.
```

Comment:
I'm using the name `ObjectSchemaProperty` here and leaving it up to the other renaming PR to modify this to `PropertySchema`.